#! /bin/sh /usr/share/dpatch/dpatch-run

## DP: Fixes DLL maps

@DPATCH@
Index: xamarin-pkg-gtk-sharp2/atk/atk-sharp.dll.config.in
===================================================================
--- xamarin-pkg-gtk-sharp2.orig/atk/atk-sharp.dll.config.in
+++ xamarin-pkg-gtk-sharp2/atk/atk-sharp.dll.config.in
@@ -1,4 +1,6 @@
 <configuration>
   <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libatk-1.0-0.dll" target="libatk-1.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="glibsharpglue-2" target="/usr/lib/cli/glib-sharp-2.0/libglibsharpglue-2.so"/>
+  <dllmap dll="atksharpglue-2" target="/usr/lib/cli/atk-sharp-2.0/libatksharpglue-2.so"/>
 </configuration>
Index: xamarin-pkg-gtk-sharp2/gdk/gdk-sharp.dll.config.in
===================================================================
--- xamarin-pkg-gtk-sharp2.orig/gdk/gdk-sharp.dll.config.in
+++ xamarin-pkg-gtk-sharp2/gdk/gdk-sharp.dll.config.in
@@ -3,4 +3,6 @@
   <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgdk-win32-2.0-0.dll" target="libgdk-@GDK_BACKEND@-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgdk_pixbuf-2.0-0.dll" target="libgdk_pixbuf-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="gdksharpglue-2" target="/usr/lib/cli/gdk-sharp-2.0/libgdksharpglue-2.so"/>
+  <dllmap dll="glibsharpglue-2" target="/usr/lib/cli/glib-sharp-2.0/libglibsharpglue-2.so"/>
 </configuration>
Index: xamarin-pkg-gtk-sharp2/glade/glade-sharp.dll.config.in
===================================================================
--- xamarin-pkg-gtk-sharp2.orig/glade/glade-sharp.dll.config.in
+++ xamarin-pkg-gtk-sharp2/glade/glade-sharp.dll.config.in
@@ -1,3 +1,4 @@
 <configuration>
   <dllmap dll="libglade-2.0-0.dll" target="libglade-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="gladesharpglue-2" target="/usr/lib/cli/glade-sharp-2.0/libgladesharpglue-2.so"/>
 </configuration>
Index: xamarin-pkg-gtk-sharp2/glib/glib-sharp.dll.config.in
===================================================================
--- xamarin-pkg-gtk-sharp2.orig/glib/glib-sharp.dll.config.in
+++ xamarin-pkg-gtk-sharp2/glib/glib-sharp.dll.config.in
@@ -2,4 +2,5 @@
   <dllmap dll="libglib-2.0-0.dll" target="libglib-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgthread-2.0-0.dll" target="libgthread-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="glibsharpglue-2" target="/usr/lib/cli/glib-sharp-2.0/libglibsharpglue-2.so"/>
 </configuration>
Index: xamarin-pkg-gtk-sharp2/gtk/gtk-sharp.dll.config.in
===================================================================
--- xamarin-pkg-gtk-sharp2.orig/gtk/gtk-sharp.dll.config.in
+++ xamarin-pkg-gtk-sharp2/gtk/gtk-sharp.dll.config.in
@@ -4,4 +4,6 @@
   <dllmap dll="libatk-1.0-0.dll" target="libatk-1.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgtk-win32-2.0-0.dll" target="libgtk-@GDK_BACKEND@-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libgdk-win32-2.0-0.dll" target="libgdk-@GDK_BACKEND@-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="gtksharpglue-2" target="/usr/lib/cli/gtk-sharp-2.0/libgtksharpglue-2.so"/>
+  <dllmap dll="glibsharpglue-2" target="/usr/lib/cli/glib-sharp-2.0/libglibsharpglue-2.so"/>
 </configuration>
Index: xamarin-pkg-gtk-sharp2/pango/pango-sharp.dll.config.in
===================================================================
--- xamarin-pkg-gtk-sharp2.orig/pango/pango-sharp.dll.config.in
+++ xamarin-pkg-gtk-sharp2/pango/pango-sharp.dll.config.in
@@ -3,4 +3,6 @@
   <dllmap dll="libgobject-2.0-0.dll" target="libgobject-2.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libpango-1.0-0.dll" target="libpango-1.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
   <dllmap dll="libpangocairo-1.0-0.dll" target="libpangocairo-1.0@LIB_PREFIX@.0@LIB_SUFFIX@"/>
+  <dllmap dll="pangosharpglue-2" target="/usr/lib/cli/pango-sharp-2.0/libpangosharpglue-2.so"/>
+  <dllmap dll="glibsharpglue-2" target="/usr/lib/cli/glib-sharp-2.0/libglibsharpglue-2.so"/>
 </configuration>
